sdk/core/azure-core-experimental/src/main/java/com/azure/core/experimental/implementation/AzureEnvironment.java

```diff
@@ -0,0 +1,352 @@
+    }
+
+    /**
+     * Provides the settings for authentication with Azure.
+     */
+    public static final AzureEnvironment AZURE = new AzureEnvironment(new HashMap<String, String>() {
+        {
+            put("portalUrl", "http://go.microsoft.com/fwlink/?LinkId=254433");
+            put("publishingProfileUrl", "http://go.microsoft.com/fwlink/?LinkId=254432");
+            put("managementEndpointUrl", "https://management.core.windows.net/");
+            put("resourceManagerEndpointUrl", "https://management.azure.com/");
+            put("sqlManagementEndpointUrl", "https://management.core.windows.net:8443/");
+            put("sqlServerHostnameSuffix", ".database.windows.net");
+            put("galleryEndpointUrl", "https://gallery.azure.com/");
+            put("activeDirectoryEndpointUrl", "https://login.microsoftonline.com/");
+            put("activeDirectoryResourceId", "https://management.core.windows.net/");
+            put("activeDirectoryGraphResourceId", "https://graph.windows.net/");
+            put("microsoftGraphResourceId", "https://graph.microsoft.com/");
+            put("dataLakeEndpointResourceId", "https://datalake.azure.net/");
+            put("activeDirectoryGraphApiVersion", "2013-04-05");
+            put("storageEndpointSuffix", ".core.windows.net");
+            put("keyVaultDnsSuffix", ".vault.azure.net");
+            put("azureDataLakeStoreFileSystemEndpointSuffix", "azuredatalakestore.net");
+            put("azureDataLakeAnalyticsCatalogAndJobEndpointSuffix", "azuredatalakeanalytics.net");
+            put("azureLogAnalyticsResourceId", "https://api.loganalytics.io/");
+            put("azureApplicationInsightsResourceId", "https://api.applicationinsights.io/");
+        }
+    });
+
+    /**
+     * Provides the settings for authentication with Azure China.
+     */
+    public static final AzureEnvironment AZURE_CHINA = new AzureEnvironment(new HashMap<String, String>() {
+        {
+            put("portalUrl", "http://go.microsoft.com/fwlink/?LinkId=301902");
+            put("publishingProfileUrl", "http://go.microsoft.com/fwlink/?LinkID=301774");
+            put("managementEndpointUrl", "https://management.core.chinacloudapi.cn/");
+            put("resourceManagerEndpointUrl", "https://management.chinacloudapi.cn/");
+            put("sqlManagementEndpointUrl", "https://management.core.chinacloudapi.cn:8443/");
+            put("sqlServerHostnameSuffix", ".database.chinacloudapi.cn");
+            put("galleryEndpointUrl", "https://gallery.chinacloudapi.cn/");
+            put("activeDirectoryEndpointUrl", "https://login.chinacloudapi.cn/");
+            put("activeDirectoryResourceId", "https://management.core.chinacloudapi.cn/");
+            put("activeDirectoryGraphResourceId", "https://graph.chinacloudapi.cn/");
+            put("microsoftGraphResourceId", "https://microsoftgraph.chinacloudapi.cn/");
+            // TODO: add resource id for the china cloud for datalake once it is defined.
```

Comment:
nit: TODOs should follow the standard `// TODO (alias): Message`

```suggestion
            // TODO (vigera): add resource id for the china cloud for datalake once it is defined.
```